WWTExplorer3d/ReferenceFrame.cs

```diff
@@ -1231,23 +1258,37 @@ public class TrajectorySample
         public double H;
         public double P;
         public double R;
-        public TrajectorySample(double time, double x, double y, double z)
+        private double factor = 1;
+        public TrajectorySample(double time, double x, double y, double z, AltUnits unit = AltUnits.Meters)
```

Comment:
Not ideal that the unit and the scale factor have to be stored per-sample, but I guess with the way that the `Position` field works there isn't really a choice?

Reply:
I think we can actually get away with removing the per-sample unit and scale factor if we convert position to a method - I'm thinking with a signature like either `public Vector3d Position(AltUnits unit)` or `public Vector3d Position(double scaleFactor)`. Looking at the references to this field, it's never accessed in a context where we don't have access to the associated reference frame.